src/formatting/expr.rs

```diff
@@ -125,7 +125,31 @@ pub(crate) fn format_expr(
         | ast::ExprKind::While(..) => to_control_flow(expr, expr_type)
             .and_then(|control_flow| control_flow.rewrite(context, shape)),
         ast::ExprKind::ConstBlock(ref anon_const) => {
-            Some(format!("const {}", anon_const.rewrite(context, shape)?))
+            let between_span = mk_sp(
+                context.snippet_provider.span_after(expr.span, "const"),
+                anon_const.value.span.lo(),
+            );
+            let anon_const_str = anon_const.rewrite(context, shape)?;
+            let combined_strs = combine_strs_with_missing_comments(
+                context,
+                "const",
+                &anon_const_str,
+                between_span,
+                shape,
+                false,
+            )?;
+
+            match context.config.brace_style() {
+                BraceStyle::AlwaysNextLine => Some(format!("{}", &combined_strs)),
```

Comment:
a question, under `AlwaysNextLine` brace style, `contains_comments` value is constant (in this case is always `false`) thus i omit it. 
Is there a reason to put it into the logic above? or the test case is not extensive enough?

Reply:
> a question, under AlwaysNextLine brace style, contains_comments value is constant (in this case is always false) thus i omit it.

So the two suggestions I provided were mutually exclusive suggestions.

One suggestion was to utilize two arms, with the first arm utilizing `contains_comments` as a guard, and in the second arm there was of course no need to utilize `contains_comments` in the arg to `combine_strs_with_missing_comments`.

The second suggestion, and the one I'd encourage, is a mutually exclusive alternative to the other suggestion, which only has a single arm that matches `AlwaysNextLine` (with no guard). In this model, the value of `contains_comments` is unknown, which is why it would need to be used for determining the value of the arg to use for the last/`can_extend` param in `combine_strs_with_missing_comments`. 

Does that make sense, or were you asking a different question?